src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+            }
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException | DataProcessingException e) {
+            throw new DataProcessingException("Can't manufacturer from DB by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturer = new ArrayList<>();
+        String getAllManufacturerRequest = "SELECT * FROM manufacturers WHERE is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllManufacturersStatement =
+                         connection.prepareStatement(getAllManufacturerRequest);
+                 ResultSet resultSet = getAllManufacturersStatement
+                         .executeQuery(getAllManufacturerRequest)) {
+            while (resultSet.next()) {
+                allManufacturer.add(getDataFromResultSet(resultSet));
+            }
+        } catch (SQLException | DataProcessingException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return allManufacturer;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacturerRequest =
+                "UPDATE manufacturers SET name = ?, country = ? "
+                        + "where id = ? AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateManufacturersStatement =
+                         connection.prepareStatement(updateManufacturerRequest)) {
+            updateManufacturersStatement.setString(1, manufacturer.getName());
+            updateManufacturersStatement.setString(2, manufacturer.getCountry());
+            updateManufacturersStatement.setLong(3, manufacturer.getId());
+            updateManufacturersStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer with this id. ID:"
+                    + manufacturer.getId(), e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteManufacturerRequest =
+                "UPDATE manufacturers SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteManufacturersStatement =
+                         connection.prepareStatement(deleteManufacturerRequest)) {
+            deleteManufacturersStatement.setLong(1, id);
+            return deleteManufacturersStatement.executeUpdate() > 0;
+        } catch (SQLException | DataProcessingException e) {
+            throw new DataProcessingException("Can't delete manufacturer from DB. id = "
+                    + id, e);
+        }
+    }
+
+    private static Manufacturer getDataFromResultSet(ResultSet resultSet) throws SQLException {
```

Comment:
it would be better to name `parseManufacturer` or something like this
```suggestion
    private static Manufacturer getDataFromResultSet(ResultSet resultSet) throws SQLException {
```